File: app/src/main/java/com/ngallazzi/highlightingviewexample/MainActivity.java
package com.ngallazzi.highlightingviewexample;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.ngallazzi.highlightingview.HighlightingView;

public class MainActivity extends AppCompatActivity {
    private HighlightingView myEnlightingLayout;
    private HighlightingView myEnlightingLayoutWithProgress;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        myEnlightingLayout = (HighlightingView) findViewById(R.id.myEnlightingLayout);
        myEnlightingLayoutWithProgress = (HighlightingView)
                findViewById(R.id.myEnlightingLayoutWithProgress);
        myEnlightingLayoutWithProgress.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!myEnlightingLayoutWithProgress.isHighlighted()){
                    new DownloadFilesTask().execute();
                }else{
                    myEnlightingLayoutWithProgress.setHighlighted(false);
                }
            }
        });
    }

    private class DownloadFilesTask extends AsyncTask<Void, Void, Long> {
        @Override
        protected void onPreExecute() {
            super.onPreExecute();
            myEnlightingLayoutWithProgress.showProgressBar();
        }

        protected Long doInBackground(Void... urls) {
            long totalSize = 0;
            try {
                Thread.sleep(2000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return totalSize;
        }

        protected void onPostExecute(Long result) {
            myEnlightingLayoutWithProgress.hideProgressBar();
            myEnlightingLayoutWithProgress.setHighlighted(true);
        }
    }
}

File: app/src/main/java/com/ngallazzi/highlightingviewexample/MainActivity.kt
package com.ngallazzi.highlightingviewexample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.ngallazzi.highlightingview.HighlightingView
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread

/**
 * HighlightingView
 * Created by Nicola on 11/5/2018.
 * Copyright © 2018 Zehus. All rights reserved.
 */


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        myEnlightingLayoutWithProgress.initExecuteBackgroundTaskOnClickListener()
        myEnlightingLayoutWithProgressTwo.initExecuteBackgroundTaskOnClickListener()
    }


    private fun startHighlightingTask(view: HighlightingView) {
        view.showProgressBar()
        doAsync {
            //Execute all the lon running tasks here
            Thread.sleep(2000)
            uiThread {
                //Update the UI thread here
                view.hideProgressBar()
                view.isHighlighted = true
            }
        }
    }

    private fun HighlightingView.initExecuteBackgroundTaskOnClickListener() {
        this.setOnClickListener {
            if (!this.isHighlighted) {
                startHighlightingTask(this)
            } else {
                this.isHighlighted = false
            }
        }
    }
}
